components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -3305,6 +3416,25 @@ void RewardsServiceImpl::OnFetchBalance(FetchBalanceCallback callback,
     if (balance->total > 0) {
       profile_->GetPrefs()->SetBoolean(prefs::kRewardsUserHasFunded, true);
     }
+
+    // Record stats.
+    double balance_minus_grant = 0;
```

Comment:
could we move this logic into p3a file and just call one function from here?

Reply:
done